Lib/posixpath.py

```diff
@@ -515,19 +515,20 @@ def relpath(path, start=None):
         raise
 
 
-# Return the longest common sub-path of the sequence of paths given as input.
+# Return the longest common sub-path of the iterable of paths given as input.
 # The paths are not normalized before comparing them (this is the
 # responsibility of the caller). Any trailing separator is stripped from the
 # returned path.
 
 def commonpath(paths):
-    """Given a sequence of path names, returns the longest common sub-path."""
+    """Given an iterable of path names, returns the longest common sub-path."""
 
-    paths = tuple(map(os.fspath, paths))
+    paths = tuple(paths)
```

Comment:
Why remove `os.fspath` calls?

Reply:
Because `splitroot()` already calls `os.fspath()`:

```none
100000 loops, best of 5: 2.84 usec per loop # without `os.fspath`
100000 loops, best of 5: 3.15 usec per loop # with `os.fspath`
# -> 1.11x slower
```

Reply:
Then maybe merge it with `splitroot()`? Check for empty iterable after calling `splitroot()`.

Reply:
Then I'll use a try-except again, because `zip()` returns `iter([])` in that case and not `iter(([], [], []))`.

Reply:
Oh, I can just separate the `map()` & `zip()` calls.